src/components/SingleProduct/CustomBreadcrumbs.tsx

```diff
@@ -8,23 +8,38 @@ export const CustomBreadcrumbs = () => {
   const { smartphone, id } = useParams();
   const allProducts = [...smartphoneData, ...laptopData];
   const currentProduct = allProducts.find((el) => id && el.id === +id);
+  const location = useLocation();
 
-  const breadcrumbItems = [
-    {
-      title: 'Homepage',
-      href: `${AppRoute?.ROOT}`,
-    },
-    {
-      title: smartphone?.[0].toUpperCase().concat(smartphone?.slice(1)),
-      href: '#',
-    },
-    {
-      title: currentProduct?.title[0]
-        .toUpperCase()
-        .concat(currentProduct.title.slice(1)),
-      href: '#',
-    },
-  ];
+  let breadcrumbItems: object[];
```

Comment:
lets rewrite it as a helper:
```
type BreadcumbItem = {
....
}
const getBreadcrumbItems = (path: string): BreadcumbItem[] => {
 return items with your logic
} 
```

This helper shall be in helpers.ts file inside component folder